code/modules/mob/living/basic/pets/pet_cult/pet_cult_ai.dm

```diff
@@ -0,0 +1,246 @@
+	var/datum/team/cult/cult_team = controller.blackboard[BB_CULT_TEAM]
+	if(isnull(cult_team))
+		return null
+
+	for(var/obj/effect/rune/raise_dead/target_rune in oview(search_range, controller.pawn))
+		controller.set_blackboard_key(BB_NEARBY_RUNE, target_rune)
+		var/mob/living/occupant = locate(/mob/living/carbon/human) in get_turf(target_rune)
+		if(isnull(occupant))
+			continue
+		if(occupant.stat != DEAD || !IS_CULTIST(occupant))
+			continue
+		return target_rune
+
+	return null
+
+/datum/ai_behavior/activate_rune
+	behavior_flags = AI_BEHAVIOR_REQUIRE_MOVEMENT | AI_BEHAVIOR_CAN_PLAN_DURING_EXECUTION | AI_BEHAVIOR_REQUIRE_REACH
+	action_cooldown = 3 SECONDS
+
+/datum/ai_behavior/activate_rune/setup(datum/ai_controller/controller, target_key)
+	. = ..()
+	var/turf/target = controller.blackboard[target_key]
+	if(isnull(target))
+		return FALSE
+	set_movement_target(controller, target)
+
+/datum/ai_behavior/activate_rune/perform(seconds_per_tick, datum/ai_controller/controller, target_key)
+	. = ..()
+	var/atom/target = controller.blackboard[target_key]
+
+	if(QDELETED(target))
+		finish_action(controller, FALSE, target_key)
+		return
+
+	var/datum/team/cult/cult_team = controller.blackboard[BB_CULT_TEAM]
+	var/mob/living/sac_human = locate(/mob/living/carbon/human) in get_turf(target)
+
+	if(isnull(sac_human) || !is_convertable_to_cult(sac_human, cult_team))
+		finish_action(controller, FALSE, target_key)
+		return
+
+	var/mob/living/basic/living_pawn = controller.pawn
+	living_pawn.melee_attack(target)
+
+	finish_action(controller, TRUE, target_key)
+	return
+
+/datum/ai_behavior/activate_rune/finish_action(datum/ai_controller/controller, success, target_key)
+	. = ..()
+	controller.clear_blackboard_key(target_key)
+
+
+///find targets that we can revive
+/datum/ai_planning_subtree/find_dead_cultist
+
+/datum/ai_planning_subtree/find_dead_cultist/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+	if((LAZYLEN(GLOB.sacrificed) - SOULS_TO_REVIVE - GLOB.sacrifices_used) < 0)
+		controller.clear_blackboard_key(BB_DEAD_CULTIST)
+		return
+
+	var/mob/living/living_pawn = controller.pawn
+
+	if(!isnull(living_pawn.pulling))
+		return
+
+	if(controller.blackboard_key_exists(BB_DEAD_CULTIST))
+		controller.queue_behavior(/datum/ai_behavior/pull_target/cult_revive, BB_DEAD_CULTIST)
+		return SUBTREE_RETURN_FINISH_PLANNING
+
+	controller.queue_behavior(/datum/ai_behavior/find_and_set/dead_cultist, BB_DEAD_CULTIST, /mob/living/carbon/human)
+
+/datum/ai_behavior/find_and_set/dead_cultist
+
+/datum/ai_behavior/find_and_set/dead_cultist/search_tactic(datum/ai_controller/controller, locate_path, search_range)
+	var/datum/team/cult/cult_team = controller.blackboard[BB_CULT_TEAM]
+	if(isnull(cult_team))
+		return null
+	var/mob/living/living_pawn = controller.pawn
+	for(var/mob/living/carbon/human/target in oview(search_range, controller.pawn))
```

Comment:
```suggestion
	for(var/mob/living//target in oview(search_range, controller.pawn))
```

The revive rune can be used on all living mobs that have the cultist antag datum.